common/djangoapps/config_models/models.py

```diff
@@ -35,33 +37,69 @@ def save(self, *args, **kwargs):
         Clear the cached value when saving a new configuration entry
         """
         super(ConfigurationModel, self).save(*args, **kwargs)
-        cache.delete(self.cache_key_name())
+        cache.delete(self.cache_key_name(*[getattr(self, key) for key in self.KEY_FIELDS]))
 
     @classmethod
-    def cache_key_name(cls):
+    def cache_key_name(cls, *args):
```

Comment:
I like that idea, and looked at implementing it, but when I look at my actual code so far, here's how it looks in practice:

```python
data = SAMLProviderData.current(inst.entity_id)
...
config = SAMLProviderConfig.current(idp_slug)
...
data_obj = SAMLProviderData.current(entity_id)
...
data = SAMLProviderData.current(self.entity_id)
...
provider = OAuth2ProviderConfig.current(backend_name)
...
provider = SAMLProviderConfig.current(idp_slug)
...
provider = OAuth2ProviderConfig.current(backend_name)
...
provider = SAMLProviderConfig.current(idp_name)
...
return SAMLProviderConfig.current(idp_name).get_config()
```

So as you can see, the arguments are pretty clear. If I were to change it to `kwargs`, it would just result in a lot of `.current(backend_name=backend_name)` which I think is perhaps less readable. The only situation where I think `kwargs` would really help would be if you had multiple keys where one or more keys was `None`. I'm guessing single keys will be more common, but open to more feedback on this.